src/pet.js

```diff
@@ -0,0 +1,78 @@
+const ageIncrease = 1;
+const hungerIncrease = 5;
+const fitnessDecrease = 3;
+const fitnessIncrease = 4;
+const fitnessMax = 10;
+
+function Pet(name) {
+	this.name = name;
+	this.age = 0;
+	this.hunger = 0;
+	this.fitness = 10;
+	this.children = [];
+}
+
+Pet.prototype = {
+	get isAlive() {
+		return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+	},
+};
+
+Pet.prototype.growUp = function () {
+	if (!this.isAlive) {
+		throw new Error('Your pet is no longer alive :(');
+	}
+	this.age += ageIncrease;
+	this.hunger += hungerIncrease;
+	this.fitness -= fitnessDecrease;
+};
+
+Pet.prototype.walk = function () {
+	if (!this.isAlive) {
+		throw new Error('Your pet is no longer alive :(');
+	}
+	if (this.fitness < fitnessMax - fitnessIncrease) {
+		this.fitness += fitnessIncrease;
+	} else {
+		this.fitness = fitnessMax;
+	}
+};
+
+Pet.prototype.feed = function () {
+	if (!this.isAlive) {
+		throw new Error('Your pet is no longer alive :(');
+	}
+	if (this.hunger > 2) {
+		this.hunger -= 3;
+	} else {
+		this.hunger = 0;
+	}
```

Comment:
Rather than checking the value before deciding what operation to perform on it, this could be shortened to:
```
this.hunger = Math.max(0, this.hunger - 3)
```